### DISCUSSION TITLE
Positron won't recognize .qmd file types when reopening project in a new session

### DISCUSSION THREAD

**Discussion by kyleGrealis** [posit-dev/positron#3713] [Q&A]
I'm using Quarto to write notes on a learning project. There weren't any previous issues with this project on the same machine while using VS Code.

The issue seems to be that Positron will not recognize the `.qmd` file type when reopening the project (as noted by the text file icon next to the filename in the included image). Also, when trying to create a new code block using typical shortcuts or even the manual way, the Quarto doc does not recognize this or any previous code cells as being anything other than as if they were `.txt` format.

Though the example code block in the image is non-executable, removing the extra `{}` does not cause it to perform as expected.

**EDIT for clarity: I experience typical functionality when creating a *new* Quarto doc. Code blocks render, document knits, markdown syntax -- all good. However, closing Positron and reopening causes this file to be viewed as `.txt` and Quarto functionality is gone.**

System specs included:

![image](https://github.com/posit-dev/positron/assets/84798289/bb23e1ca-2783-44ce-bbb9-a43fa6b6fbdb)

Thank you all for making such a great tool. R with VS Code was clunky and this was great to see!

**Comment by petetronic** [comment]
The icon for the fast-api-notes.qmd file in the Explorer pane, and the missing Preview button on the Editor bar seems to me that the Quarto extension is not installed or disabled somehow? Can you check your extensions to ensure the Quarto extension is installed and enabled?

Positron pre-release [2024.06.1-27](https://github.com/posit-dev/positron/releases/tag/2024.06.1-27) did not have the Quarto extension bundled, but we should have it included in an upcoming pre-release soon.

**Comment by kyleGrealis** [comment]
![image](https://github.com/posit-dev/positron/assets/84798289/83447d7d-b175-4494-8cee-a4c90dfcead6)

![image](https://github.com/posit-dev/positron/assets/84798289/ba7cc1ca-3baa-48e8-872e-05216364915d)

I ensured that the Quarto extension and latest version of Quarto were installed prior to opening the issue and confirmed again just now. Thank you for replying. Please let me know if there are other things you'd like me to try or replicate.

**Reply by kyleGrealis** [reply]
Adding another screenshot showing lack of Preview button in the editor:
![image](https://github.com/posit-dev/positron/assets/84798289/880bb47d-d682-46ac-a256-2ba59cb35171)

**Reply by kyleGrealis** [reply]
Cannot replicate for the same project on my Windows machine:

![image](https://github.com/posit-dev/positron/assets/84798289/dd7d3c90-974c-4785-8efd-ae0b9b8ce491)

**Reply by petetronic** [reply]
Thank you for checking that and the extra info! Let me get the team to try and reproduce this on Ubuntu 24.04. There might also be relevant errors logged in the Output view in the "Extension Host" channel?

**Reply by kyleGrealis** [reply]
Where can I find that?

**Reply by petetronic** [reply]
By default, the Output view is docked as a tab in the Panel. If the Panel tabs are tight on space, it may be hidden by the `...` ellipsis (after CONSOLE | TERMINAL | PROBLEMS | ...). If it is not there, there is a command palette option `Output: Focus on Output View` to bring it into focus. Once that view is open, there is a drop down of log output "channels". YMMV, but the `Extension Host` channel may list issues that were encountered starting the Quarto extension.

**Comment by kyleGrealis** [comment]
I have resolved the issue and (surprise!) it's more user error than Positron. I had the following configuration I had ported over from my VS Code settings.json:
```
"files.associations": {
        "*.qmd": "Quarto",
        "*.R": "R"
},
```
Removing these resolved the issue and Quarto works as expected.

**Reply by petetronic** [reply]
Thanks for tracking that down @kyleGrealis! I ended up getting an x86_64 Ubuntu 24.04 VM running and was able to use Quarto with Positron. Glad you worked it out!
